tests/state/ViewerState/time_slice.test.ts

```diff
@@ -104,11 +104,10 @@ describe("useViewerStateStore: TimeSlice", () => {
     await act(async () => {
       await result.current.setFrame(3);
     });
-    expect(result.current.currentFrame).toBe(3);
     expect(result.current.pendingFrame).toBe(3);
+    expect(result.current.currentFrame).toBe(3);
 
     await setDatasetAsync(result, MOCK_DATASET_WITH_TWO_FRAMES);
-    expect(result.current.currentFrame).toBe(1);
```

Comment:
I may have missed it, but what changed in the code that necessitates this test change?

Reply:
Ah! Sorry, should have included a comment on this. This test would fail sometimes due to a race condition, since `currentFrame` is set asynchronously. `pendingFrame` still represents the intended time value that the state is loading.

I could also fix this by adding `await sleep(10)` or some other  async timing mechanism.

Reply:
Gotcha! Thanks for the explanation, I'll leave it up to you whether you want to add some sort of timing mechanism to fix it or not, based on how well tested you think the intended behavior is in other tests

Reply:
I'll leave it for now... there's another PR where I could probably insert some adjustments to the time slice tests.